python/ray/data/read_api.py

```diff
@@ -1755,6 +1761,109 @@ def create_connection():
     )
 
 
+@PublicAPI(stability="alpha")
+def read_databricks_uc_tables(
+    *,
+    warehouse_id: str,
+    table: Optional[str] = None,
+    query: Optional[str] = None,
+    catalog: Optional[str] = None,
+    schema: Optional[str] = None,
+    parallelism: int = -1,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+) -> Dataset:
+    """
+    Read from a Databricks UC table or Databricks SQL execution result that queries
+    from Databricks UC tables.
+    If it is not called in databricks runtime, you need to set environment varaibles
+    'DATABRICKS_HOST' and 'DATABRICKS_TOKEN' firstly.
+
+    This reader is implemented based on
+    [Databricks statemenet execution API](https://docs.databricks.com/api/workspace/statementexecution).
+
+    Examples:
+        >>> from ray.data.read_api import read_databricks_uc_tables
+        >>>
+        >>> ds = read_databricks_uc_tables(
+        ...   warehouse_id='a885ad08b64951ad',
+        ...   catalog='catalog_1',
+        ...   schema='db_1',
+        ...   query='select id from table_1 limit 750000',
+        ...   parallelism=2,
+        ... )
+
+    Args:
+        warehouse_id: The id of the databricks warehouse, the query statement is
+            executed on this warehouse.
+        table: The name of UC table you want to read. If this argument is set,
+            you can't set 'query' argument, and the reader generates query
+            of 'select * from {table_name}' under the hood.
+        query: The query you want to execute. If this argument is set,
+            you can't set 'table_name' argument.
+        catalog: (Optional) The default catalog name used by the query
+        schema: (Optional) The default schema used by the query
+        parallelism: The requested parallelism of the read. Defaults to -1,
+            which automatically determines the optimal parallelism for your
+            configuration. You should not need to manually set this value in most cases.
+            For details on how the parallelism is automatically determined and guidance
+            on how to tune it, see :ref:`Tuning read parallelism
+            <read_parallelism>`.
+        ray_remote_args: kwargs passed to :meth:`~ray.remote` in the read tasks.
+
+    Returns:
+        A :class:`Dataset` containing the queried data.
+    """
+    from ray.data.datasource.databricks_uc_datasource import DatabricksUCDatasource
+    from ray.util.spark.databricks_hook import get_dbutils
+    from ray.util.spark.utils import get_spark_session, is_in_databricks_runtime
```

Comment:
Any reason we can't import this at the top level?

Reply:
it will import pyspark package, I hope we can lazily import `pyspark` , importing pyspark package is slow.

Reply:
Gotcha, makes sense